stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/dao/ChannelConfigDao.kt

```diff
@@ -4,19 +4,37 @@ import androidx.room.Dao
 import androidx.room.Insert
 import androidx.room.OnConflictStrategy
 import androidx.room.Query
+import androidx.room.Transaction
 import io.getstream.chat.android.livedata.entity.ChannelConfigEntity
+import io.getstream.chat.android.livedata.entity.ChannelConfigInnerEntity
+import io.getstream.chat.android.livedata.entity.CommandInnerEntity
 
 @Dao
 internal interface ChannelConfigDao {
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    suspend fun insertMany(channelConfigEntities: List<ChannelConfigEntity>)
+    @Transaction
+    suspend fun insert(channelConfigEntities: List<ChannelConfigEntity>) {
+        channelConfigEntities.map(ChannelConfigEntity::channelConfigInnerEntity).also { insertConfigs(it) }
+        channelConfigEntities.flatMap(ChannelConfigEntity::commands).also { insertCommands(it) }
+    }
+
+    @Transaction
+    suspend fun insert(channelConfigEntity: ChannelConfigEntity) {
+        insertConfig(channelConfigEntity.channelConfigInnerEntity)
+        insertCommands(channelConfigEntity.commands)
+    }
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertConfig(channelConfigInnerEntity: ChannelConfigInnerEntity)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertConfigs(channelConfigInnerEntities: List<ChannelConfigInnerEntity>)
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    suspend fun insert(channelConfigEntity: ChannelConfigEntity)
+    suspend fun insertCommands(commands: List<CommandInnerEntity>)
```

Comment:
These should never be called directly, outside of this Dao, right? We could make the Dao an abstract class and mark these methods as `protected` to avoid someone accidentally calling these in the future.

Reply:
Good idea